devito/ir/stree/algorithms.py

```diff
@@ -51,7 +51,7 @@ def st_schedule(clusters):
 
         # The reused sub-trees might acquire some new sub-iterators
         for i in pointers[:index]:
-            mapper[i].ispace = IterationSpace.merge(mapper[i].ispace,
+            mapper[i].ispace = IterationSpace.union(mapper[i].ispace,
                                                     c.ispace.project([i.dim]))
         # Later sub-trees, instead, will not be used anymore
```

Comment:
What about later sub-trees? Do you mean following? I do not quite get it?
Does it mean next trees or do you refer to a later part of code?

Reply:
I think I actually meant "nested"

Reply:
fixed